SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -102,6 +129,21 @@ void CSpeciesFlameletSolver::Preprocessing(CGeometry* geometry, CSolver** solver
                   SU2_MPI::GetComm());
   if ((rank == MASTER_NODE) && (n_not_in_domain_global > 0))
     cout << "Number of points outside manifold domain: " << n_not_in_domain_global << endl;
+
+  /*--- Compute preferential diffusion scalar gradients. ---*/
+  if (config->GetPreferentialDiffusion()) {
+    switch (config->GetKind_Gradient_Method_Recon()) {
+      case GREEN_GAUSS:
+        SetAuxVar_Gradient_GG(geometry, config);
+        break;
+      case WEIGHTED_LEAST_SQUARES:
+      case LEAST_SQUARES:
+        SetAuxVar_Gradient_LS(geometry, config);
+        break;
+      default:
+        break;
+    }
+  }
```

Comment:
Hi Evert, 

I hope you are doing well, 
Thank you so much for this pull request, I am currently using it for simulating premixed laminar flames in a perforated burners. I have a question regarding this piece of code, are the gradients of the preferential diffusion terms only available when the MUSCL_SPECIES=YES and the NUM_METHOD_GRAD_RECON is GREEN_GAUSS or LEAST_SQUARES? 
I am wondering if config->GetKind_Gradient_Method_Recon() should be config->GetKind_Gradient_Method() and the WEIGHTED_LEAST_SQUARES and LEAST_SQUARES should be switched in the order, like 
case LEAST_SQUARES:
case WEIGHTED_LEAST_SQUARES:
       ........

Let me know what you think about it and thank you so much in advance!!!!.

Reply:
The gradients of the preferential diffusion terms are available with or without MUSCL enabled. I'll try your suggestion and see whether it affects the 3D flame shape with preferential diffusion enabled.